Fixes sdql2 applying comparison operators to lists incorrectly

MrsTonedOne opened the pull request:
if result is already assigned a list, doing things like `+=` or `&=` would change the list by reference. this would cause things like WHERE var & `something` to behave incorrectly if var was a list, actually changing the list by removing every thing that wasn't `something` rather than creating a new list with everything that matches

@kevinz000

MrsTonedOne committed:
Fixes sdql2 applying comparison operators to lists incorrectly

MrStonedOne commented:
want to 100% fuck up everything?

`select /atom/ in world where contents & "blahblahblah"

xxalpha commented:
you can't be serious with this

MrStonedOne commented:
I really wish i wasn't.

`SELECT /mob/living/carbon/human IN world`

Works fine

`SELECT /mob/living/carbon/human IN world where contents & "blah"` stupid, useless, there isn't actually any reason to do this query since an empty list passes the check, so it would return all the humans, BUT, it will also apply an &= operation on the contents list, causing it to get emptyed, meaning every human would have a blank contents list.

doing result = contents & "blah" creates a new list, doing result &= "blah" applies the operation to the list contained in result.

This is a lesson i learned with my VV refactor.

RemieRichards commented on code/modules/admin/verbs/SDQL2/SDQL_2.dm:
```diff
@@ -304,19 +304,19 @@
 			switch(op)
 				if("+")
-					result += val
+					result = (result + val)
```
just comment these, so some enterprising future coder doesn't try to get smart or fresh and "fix" them.

MrStonedOne replied:
na, I like the idea of them learning this lesson the hard way.

There should really be a ban on those kinds of changes anyways.

xxalpha replied:
Please do comment, our reviewers don't pay enough attention to these small details. In a year someone will fix this and it will get merged.

MrStonedOne replied:
I'm fucking sick and tired of this shit where people touch code they don't understand.

By adding the comment, you imply that every bit of code thats the same without the same comment is safe to change, but thats not true.

These landmines are important for teaching new coders not to fuck with shit they don't understand.

When you see something and go "why is it doing it that way". the answer is not to go "fuck it, it seems to still work when i change it" the answer is to leave it the fuck alone until it doing it that way is actually an issue.

How many god damn bugs have we had this last year from non-changes like that. I'm sick of them.

optimumtact replied:
Ahh fuck it I can't care enough about this to bother

MrPerson replied:
>How many god damn bugs have we had this last year from non-changes like that. I'm sick of them.
 
And if  you left the fucking comment, guess how often we'd have to deal with this in the future? 0

MrStonedOne replied:
Or, how about, the next time somebody makes a pr where they make syntax changes for no fucking reason, you all fucking close it.

optimumtact committed:
Add a comment to prevent future reverts

Just in case

MrsTonedOne committed:
Removes oranges comment

I'm fucking sick and tired of this shit where people touch code they don't understand.

By adding the comment, you imply that every bit of code thats the same without the same comment is safe to change, but thats not true.

These landmines are important for teaching new coders not to fuck with shit they don't understand.

When you see something and go "why is it doing it that way". the answer is not to go "fuck it, it seems to still work when i change it" the answer is to leave it the fuck alone until it doing it that way is actually an issue.

How many god damn bugs have we had this last year from non-changes like that. I'm sick of them.
